sdk/clientcore/annotation-processor-test/src/main/java/io/clientcore/annotation/processor/test/SimpleXmlSerializableServiceImpl.java

```diff
@@ -178,7 +200,25 @@ private static Object decodeNetworkResponse(BinaryData data, ObjectSerializer se
             }
             return serializer.deserializeFromBytes(data.toBytes(), token);
         } catch (IOException e) {
-            throw LOGGER.logThrowableAsError(CoreException.from(e));
+            LOGGER.atWarning().setThrowable(e).log(() -> "Failed to deserialize the error entity.");
+            return null;
+        }
+    }
+
+    private static HttpResponseException instantiateUnexpectedException(int responseCode, Response<BinaryData> response, BinaryData data, Object decodedValue) {
+        StringBuilder exceptionMessage = new StringBuilder("Status code ").append(responseCode).append(", ");
+        String contentType = response.getHeaders().getValue(HttpHeaderName.CONTENT_TYPE);
+        if ("application/octet-stream".equalsIgnoreCase(contentType)) {
+            String contentLength = response.getHeaders().getValue(HttpHeaderName.CONTENT_LENGTH);
+            exceptionMessage.append("(").append(contentLength).append("-byte body)");
+        } else if (data == null || data.toBytes().length == 0) {
+            exceptionMessage.append("(empty body)");
+        } else {
+            exceptionMessage.append('"').append(new String(data.toBytes(), StandardCharsets.UTF_8)).append('"');
```

Comment:
@alzimmermsft I have been considering if we want to port this functionality over from RestProxy, especially where we check the content type and length to determine body length, as well as the exception messages. Should we just let it fall through instead? I think we will still need to check if data is null/empty to avoid decoding it, but for other cases, just let it fall through. Also, do we need custom messaging? It doesn't add much.

Reply:
The checks around `application/octet-stream` were done as there isn't a good string representation as it's expected to just be a stream of bytes, which could be anything.